### PR TITLE
zk: implement ZkList command

### PR DESCRIPTION
**Pull request by WhyNotHugo**
Surfaces the `zk.list` LSP command in the same way as `zk.new` and `zk.index`.

### CODE REVIEW

File: lua/lspconfig/server_configurations/zk.lua [hunk 1]
```diff
@@ -1,5 +1,13 @@
 local util = require 'lspconfig.util'
 
+local function find_zk_ancestor(startpath)
+  return util.search_ancestors(startpath, function(path)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/zk.lua:4]: could you use vim.fs moudle replace util module here. It's helpful when we want remove some fs util functions in util.lua in future.

Comment (WhyNotHugo) [reply lua/lspconfig/server_configurations/zk.lua]: Updated.

File: lua/lspconfig/server_configurations/zk.lua [hunk 2]
```diff
@@ -1,5 +1,13 @@
 local util = require 'lspconfig.util'
 
+local function find_zk_root(startpath)
+  for dir in vim.fs.parents(startpath) do
+    if vim.fn.isdirectory(dir .. '/.zk') == 1 then
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/zk.lua:5]:
```suggestion
    if vim.fn.isdirectory(vim.fs.joinpath(dir, '.zk')) == 1 then
```

**Comment by WhyNotHugo** [comment]
Thanks for the review

**Pull request merged**